src/utilities/transformSchema.js

```diff
@@ -0,0 +1,298 @@
+/**
+ * Copyright (c) 2015-present, Facebook, Inc.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @flow
+ */
+
+import type { ObjMap } from '../jsutils/ObjMap';
+import keyValMap from '../jsutils/keyValMap';
+import invariant from '../jsutils/invariant';
+import type { GraphQLSchemaConfig } from '../type/schema';
+import { GraphQLSchema } from '../type/schema';
+import { isSpecifiedScalarType } from '../type/scalars';
+import { isIntrospectionType } from '../type/introspection';
+import type { GraphQLDirectiveConfig } from '../type/directives';
+import { GraphQLDirective } from '../type/directives';
+import { GraphQLList, GraphQLNonNull } from '../type/wrappers';
+import type {
+  GraphQLNamedType,
+  GraphQLScalarTypeConfig,
+  GraphQLObjectTypeConfig,
+  GraphQLInterfaceTypeConfig,
+  GraphQLUnionTypeConfig,
+  GraphQLEnumTypeConfig,
+  GraphQLInputObjectTypeConfig,
+} from '../type/definition';
+import {
+  GraphQLScalarType,
+  GraphQLObjectType,
+  GraphQLInterfaceType,
+  GraphQLUnionType,
+  GraphQLEnumType,
+  GraphQLInputObjectType,
+  isListType,
+  isNonNullType,
+  isScalarType,
+  isObjectType,
+  isInterfaceType,
+  isUnionType,
+  isEnumType,
+  isInputObjectType,
+} from '../type/definition';
+
+type ExactSchemaConfig = $Exact<GraphQLSchemaConfig> & {
+  types: Array<*>,
+  directives: Array<*>,
+};
+
+type ExactDirectiveConfig = $Exact<GraphQLDirectiveConfig> & {
+  args: ObjMap<*>,
+};
+
+type ExactScalarTypeConfig = $Exact<GraphQLScalarTypeConfig<*, *>>;
+
+type ExactObjectTypeConfig = $Exact<GraphQLObjectTypeConfig<*, *>> & {
+  fields: () => ObjMap<*>,
+  interfaces: () => Array<*>,
+  extensionASTNodes: $ReadOnlyArray<*>,
+};
+
+type ExactInterfaceTypeConfig = $Exact<GraphQLInterfaceTypeConfig<*, *>> & {
+  fields: () => ObjMap<*>,
+  extensionASTNodes: $ReadOnlyArray<*>,
+};
+
+type ExactUnionTypeConfig = $Exact<GraphQLUnionTypeConfig<*, *>> & {
+  types: () => Array<*>,
+};
+
+type ExactEnumTypeConfig = $Exact<GraphQLEnumTypeConfig>;
+
+type ExactInputObjectTypeConfig = $Exact<GraphQLInputObjectTypeConfig> & {
+  fields: () => ObjMap<*>,
+};
+
+type TypeTransformer = {
+  Schema?: (config: ExactSchemaConfig) => GraphQLSchema,
+  Directive?: (config: ExactDirectiveConfig) => GraphQLDirective,
+  ScalarType?: (config: ExactScalarTypeConfig) => GraphQLScalarType,
+  ObjectType?: (config: ExactObjectTypeConfig) => GraphQLObjectType,
+  InterfaceType?: (config: ExactInterfaceTypeConfig) => GraphQLInterfaceType,
+  UnionType?: (config: ExactUnionTypeConfig) => GraphQLUnionType,
+  EnumType?: (config: ExactEnumTypeConfig) => GraphQLEnumType,
+  InputObjectType?: (
+    config: ExactInputObjectTypeConfig,
+  ) => GraphQLInputObjectType,
+};
+
+/**
+ * Note: This class is not a part of public API and is a subject to changes
+ * in the future.
+ */
+export class SchemaTransformer {
```

Comment:
I'm not sure why this is a class since the only think you can do with it is `instance.transformSchema()`. I'd suggest refactoring this to just export a function `transformSchema(schema, transformer)` since that presents the same behavior in a simpler API